x-pack/platform/packages/shared/onechat/onechat-server/src/tools.ts

```diff
@@ -174,13 +174,20 @@ export type ExecutableToolHandlerFn<TParams = Record<string, unknown>, TResult =
   params: ExecutableToolHandlerParams<TParams>
 ) => Promise<RunToolReturn<TResult>>;
 
+/**
+ * Return value for {@link ToolHandlerFn} / {@link RegisteredTool}
+ */
+export interface ToolHandlerReturn<T = unknown> {
+  result: T;
+}
+
 /**
  * Tool handler function for {@link RegisteredTool} handlers.
  */
 export type ToolHandlerFn<
   TParams extends Record<string, unknown> = Record<string, unknown>,
   RunOutput = unknown
-> = (args: TParams, context: ToolHandlerContext) => MaybePromise<RunOutput>;
+> = (args: TParams, context: ToolHandlerContext) => MaybePromise<ToolHandlerReturn<RunOutput>>;
```

Comment:
Changing the return type of `ToolHandlerFn` to support returning structured data (and not just the "plain" result)

Reply:
How is this different from "plain" result? The nested `result` still infers type from `RunOutput` ? We just added one level of nesting that doesn't change anything for now right?

Reply:
That doesn't change anything for now, that's correct. It prepares to the option of having tools returning "more" than just their plain result, such as the corresponding citations, following a pre-defined contract (via that `ToolHandlerReturn` type). I'm doing it now because it's a breaking change of contract, so better doing it before we have more tools.